rasterio/_warp.pyx

```diff
@@ -361,10 +387,14 @@ def _reproject(
     cdef GDALTransformerFunc pfnTransformer = NULL
     cdef GDALWarpOptions *psWOptions = NULL
 
+    # Set up GDALCreateGenImgProjTransformer2 keyword arguments.
+    cdef char **imgProjOptions = NULL
+    CSLSetNameValue(imgProjOptions, "GCPS_OK", "TRUE")
```

Comment:
I'm assuming that setting `GCPS_OK` to True will just fall back on using the srs transform?

@dnomadb brought up that there might be some additional configuration on the user's end: particularly **configuring how the gcps are fit** (polynomial order? thin plate spline?) and specifying a threshold for the tolerance. These appear to be available in (GDALCreateGenImgProjTransformer2)[http://www.gdal.org/gdal__alg_8h.html#a94cd172f78dbc41d6f407d662914f2e3] but how to expose them to Rasterio users?

One thing I did not find in the GDAL docs is how to retrieve the **overall RMSE** of the returned transformer. That's a fairly common metric for evaluating the fit and quality of the output so I'd be surprised if there weren't some way to access it.